sheets/utils.py

```diff
@@ -40,407 +40,194 @@ def generate_google_client_config():
-            raw_row_data (iterable of str): The raw row data
+            row_data (iterable): Cell values from a row in a sheet
 
         Returns:
-            list of str: The row data containing only columns with user-entered values
+            list: Values for columns that contain data entered by a user in a form
         """
         return [
-            col
-            for i, col in enumerate(raw_row_data)
-            if i not in coupon_request_sheet_spec.calculated_column_indices
+            col for i, col in enumerate(row_data) if i in self.form_input_column_indices
         ]
 
-    def get_product_id(self):
-        """
-        Gets the id for the most recently-created product associated with the CourseRun/Program indicated
-        by this row.
 
-        Returns:
-            int: The most recently-created product ID for the object indicated by the text ID in the spreadsheet
-        """
-        for product_object_cls in [CourseRun, Program]:
-            product_object = (
-                product_object_cls.objects.live()
-                .with_text_id(self.product_text_id)
-                .prefetch_related("products")
-                .first()
-            )
-            if product_object:
-                break
-        product = (
-            None
-            if not product_object
-            else product_object.products.order_by("-created_on").first()
-        )
-        if not product_object:
-            raise InvalidSheetProductException(
-                "Could not find a CourseRun or Program with text id '%s'"
-                % self.product_text_id
-            )
-        elif not product:
-            raise InvalidSheetProductException(
-                "No Products associated with %s" % str(product_object)
-            )
-        return product.id
-
-
-class CouponAssignmentRow:
-    """Represents a row of a coupon assignment sheet"""
-
-    def __init__(
-        self, row_index, code, assignee_email, status, status_date
-    ):  # pylint: disable=too-many-arguments
-        self.row_index = row_index
-        self.code = code
-        self.email = assignee_email
-        self.status = status
-        self.status_date = status_date
-
-    @classmethod
-    def parse_raw_data(cls, row_index, raw_row_data):
-        """
-        Parses raw row data
+class WatchableSheetMetadata(SheetMetadata):
+    """Metadata for a type of Google Sheet that this app interacts with and tracks via file watch/webhook"""
 
-        Args:
-            row_index (int): The row index according to the spreadsheet (not zero-based)
-            raw_row_data (list of str): The raw row data
+    sheet_file_id = None
 
-        Returns:
-            CouponAssignmentRow: The parsed data row
+    @property
+    def handler_url_stub(self):
+        """The URL stub that should be used by Google when a change is made to a watched file"""
+        return "{}?sheet={}".format(
+            reverse("handle-watched-sheet-update"), self.sheet_type
```

Comment:
Can we use `urllib.parse.quote_plus` for `self.sheet_type`, just in case there is a value in the future with spaces or other escapable characters?